webapp/src/containers/MasternodesPage/saga.tsx

```diff
@@ -115,8 +113,20 @@ export function* handleRestartNode() {
       const data = yield call(getAddressInfo, masternodeOwner);
       if (data.ismine && !data.iswatchonly) {
         const updatedConf = yield call(getConfigurationDetails);
-        updatedConf.masternode_operator = masternodeOperator;
-        updatedConf.masternode_owner = masternodeOwner;
+        updatedConf[network] = {
+          ...updatedConf[network],
+          masternode_operator: updatedConf[network].masternode_operator
+            ? [...updatedConf[network].masternode_operator, masternodeOperator]
+            : [masternodeOperator],
+          spv: 1,
+          gen: 1,
+        };
+        if (network === TEST) {
+          updatedConf[network] = {
+            ...updatedConf[network],
+            spv_testnet: 1,
+          };
+        }
         yield put(restartModal());
         yield call(shutDownBinary);
         yield call(restartNode, { updatedConf });
```

Comment:
For this method, do you think it's a good idea to use `restartNodeSync` instead? Basically, the difference is it waits for all processes to close before proceeding. I believe restartNode can cause reindex issues on some cases. Just a suggestion

Reply:
@monstrobishi you suggest?